adventure.py

```diff
@@ -26,10 +28,24 @@
 else:
     with_colors = False
 
-if platform.system() != "Windows":
+if platform.system() != "Windows": # compatibility fixes
     clear_command = "clear"
 else:
     clear_command = "cls"
+
+with_custom = input("Generate map or use custom?") # asking user to either generate or use custom map
+if with_custom.lower() in "generate":
+    dim = int(input("Tile dimension?\n"))  # getting world dimensions from user
+    world = world_tile(dim, "world", with_colors, False,"") # creating "world" object in "table" class with user input
+    system(clear_command) # clearing screen to prepare for game
+else:
+    dim = 5
+    filename = input("Input name of file to be imported (including file extension):\n")
```

Comment:
Make sure to sanitize this input, because otherwise they could try and import other things. Maybe have it prepend a subdirectory name to the input and then store all the loadable worlds in that subdirectory, thus preventing them from loading non-loadable files? Consider also finding a way to list all the loadable files before prompting the user so that they know which options they have.

Reply:
If they import other things, then the program will throw an error because of how the open command and handler work, so arbitrary code execution isn't a concern. I've already added an item to the project board for the other things, and will look into creating a directory for all the other files and then printing it-- but that's not a crucial feature at this point. Remember, anyone using this is going to be doing so for fun, not to break into it-- it's not a server, and the code is open-source anyways.